docs/modules/candid-guide/index.adoc

```diff
@@ -0,0 +1,1098 @@
+
+The https://github.com/dfinity/candid/tree/master/tools/ui[Candid UI] creates a Web interface from the Candid interface of any canister.
+
+
+[#how-do-i]
+== How do I…
+
+Now that you have a good overview of Candid and its various moving parts, this section gives you concrete instructions to achieve certain tasks.
+
+=== Interact with canisters on the command line?
+
+The `dfx` tool provides the `dfx canister call` command, which understands <<textual-values,Candid textual values>>:
+....
+dfx canister call msgt2-6nlzx-xq add_entry ("John Doe", record { street = "Hauptstraße", city = "Zürich" })
+....
+
+See TODO to read more about how to use the `dfx` tool.
+
+=== Interact with canisters via Candid UI?
+
+=== Use Candid as a Motoko canister developer?
+
+If you are writing a service from scratch in Motoko, then you need to take no special actions: The Motoko compiler will translate the signature of your Canister’s top-level actor or actor class into a Candid description, and the `dfx` build tool will make sure it ends up where it needs to be.
+
+If you want to implement a _specific_ interface, for example because you want to interact with a service that expects your canister to have that interface, you can consult the 
+reference below to figure which Motoko types to use to achieve this effect. In the future, this will be simplified.
+
+=== Use Candid as Rust canister developer?
+
+TODO: Someone who implements rust canisters should write this section.
+
+=== Use Candid as a JavaScript front-end developer?
+
+TODO: Someone who implements Frontends should write this section
+
+=== Implement Candid?
+
+If you want to create a new Candid implementation, this document will not suffice. In this case, consult the https://github.com/dfinity/candid/blob/master/spec/Candid.md[Candid specification].
+
+Additionally, the official https://github.com/dfinity/candid/tree/master/test[Candid test data] is useful to make sure your implementation is compatible with Candid, even in slightly more obscure corner cases.
+
+[#reference]
+== Reference
+
+This section lists all the types supported by Candid, including their type syntax, the syntax for the textual representation, their upgrading rules and the corresponding types in Rust, Motoko and Javascript.
+
+TIP: Subtypes are the types you can change your method _results_ to, and supertypes are the types that you can change your method _arguments_ to.
+
+We only list the particular subtypes and supertypes in each section, and do not repeat every time that that the type `empty` is a subtype of any type, and that the types `reserved` and `opt t` are supertypes of any types. See the sections on these types for more details.
+
+[#type-text]
+=== Type text
+
+The `text` type is used for human readable text. More precisely, its values are sequences of unicode code points (excluding surrogate parts).
```

Comment:
```suggestion
The `text` type is used for human readable text. More precisely, its values are sequences of unicode code points (excluding surrogate parts).
```
is that parts or pairs?

Reply:
it’s “part” in https://github.com/dfinity/candid/blob/master/spec/Candid.md – I guess even a part is forbitten, let alone two that form a pair